tests/e2e/specs/wcpay/shopper/shopper-checkout-free-cart.spec.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+
+const {
+	shopper,
+	merchant,
+	clearAndFillInput,
+	uiUnblocked,
+} = require( '@woocommerce/e2e-utils' );
+
+/**
+ * Internal dependencies
+ */
+import {
+	setupCheckout,
+	fillCardDetails,
+	fillCardDetailsWCB,
+} from '../../../utils/payments';
+import { shopperWCP, merchantWCP, checkPageExists } from '../../../utils';
+
+const productName = config.get( 'products.simple.name' );
+const billingDetails = config.get( 'addresses.customer.billing' );
+const card = config.get( 'cards.basic' );
+
+describe( 'Checkout with free Cart & after Cart updates on Checkout page', () => {
+	beforeAll( async () => {
+		try {
+			await checkPageExists( 'checkout-wcb' );
+		} catch ( error ) {
+			await merchant.login();
+			await merchantWCP.addNewPageCheckoutWCB();
+			await merchant.logout();
+		}
+	} );
+
+	beforeEach( async () => {
+		await shopper.goToShop();
+		await shopper.addToCartFromShopPage( productName );
+		await shopper.goToCart();
+		await clearAndFillInput( '#coupon_code', 'free' );
+		await expect( page ).toClick( 'button[name="apply_coupon"]' );
+	} );
+
+	afterAll( async () => {
+		// Clear the cart at the end so it's ready for another test
+		await shopperWCP.emptyCart();
+	} );
+
+	it( 'checkout a free cart', async () => {
+		await shopper.goToCheckout();
+		await shopper.fillBillingDetails( billingDetails );
+		await page.waitFor( 1000 );
+		await uiUnblocked();
+		await shopper.placeOrder();
+		await expect( page ).toMatch( 'Order received' );
+	} );
+
+	it( 'update a free cart via AJAX and checkout with WCPay', async () => {
+		await shopper.goToCheckout();
+		await page.click( '.woocommerce-remove-coupon' );
+		await uiUnblocked();
+		await setupCheckout( billingDetails );
+		await fillCardDetails( page, card );
+		await shopper.placeOrder();
+		await expect( page ).toMatch( 'Order received' );
+	} );
+
+	it( 'update a free cart via AJAX and checkout with WCPay Blocks', async () => {
```

Comment:
"WCPay Blocks" does not sound right, may use the "Blocks checkout"? 

```suggestion
	it( 'remove the free coupon, can still checkout with Blocks checkout', async () => {
```



Also, I think for testing blocks checkout, we should move tests to this folder `tests/e2e/specs/blocks/shopper`? @deepakpathania may consult on this.

Reply:
Yeah, makes sense. We could then move the `goToCheckout` step to `beforeEach` as well for the rest of the tests.

Reply:
That would mix up this particular scenario ( updating a free cart to non-free) with the general flow tests.

Reply:
Improved all test descriptions.